### ISSUE TITLE
can't load remote images

### ISSUE THREAD

**Issue by makc** [meltingice/CamanJS#184]
with cors headers it should be perfectly possible to load them, why force people to use php proxies?

**Comment by makc** [comment]
basically, somewhere around [here](https://github.com/meltingice/CamanJS/blob/master/src/core/caman.coffee#L285) you need to set img.[crossorigin](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-crossorigin) (I am using canvas + image path here).

**Comment by jocoolie** [comment]
IIRC, Canvas as implemented has a "tainted" data property that is enforced by the browser and can't be changed by anything. If the image is considered tainted, Caman can't do anything on the canvas. See here: http://stackoverflow.com/questions/13674835/canvas-tainted-by-cross-origin-data and http://stackoverflow.com/questions/7129178/browser-canvas-cors-support-for-cross-domain-loaded-image-manipulation?rq=1

**Comment by makc** [comment]
yes well image will not be considered tainted if you attempt to authenticate and CORS headers say it is okay to use it. here is the workaround I am using atm:
```javascript
		var image = new Image ();
		image.crossOrigin = 'anonymous';
		image.onload = function () {

			// camanjs will replace image or canvas with new canvas in its parent
			var div = document.createElement ('div');
			div.appendChild (image);

			Caman(image, function () {
				// use selected preset, or do whatever you want here
				this [preset] ();

				this.render (function () {
					var dataUrl = div.querySelector ('canvas').toDataURL ('image/jpeg', 0.6);
					.........
				});
			});

		};
		image.src = .....;

```

**Comment by jocoolie** [comment]
Yes, that makes sense and was my next suggestion - load the image first,
then run Caman on it.

It would be good to see this functionality in the core though.
